web/src/state_data.ts

```diff
@@ -22,7 +22,29 @@ export type GroupPermissionSetting = z.output<typeof group_permission_setting_sc
 
 export const narrow_term_schema = z.object({
     negated: z.optional(z.boolean()),
-    operator: z.string(),
+    operator: z.enum([
+        "", // Used for search suggestions.
+        "channel",
+        "channels",
+        "dm",
+        "dm-including",
+        "from",
+        "group-pm-with",
+        "has",
+        "id",
+        "in",
+        "is",
+        "near",
+        "pm",
+        "pm-including",
+        "pm-with",
+        "search",
+        "sender",
+        "stream",
+        "streams",
+        "topic",
+        "with",
+    ]),
     operand: z.string(),
 });
 export type NarrowTerm = z.output<typeof narrow_term_schema>;
```

Comment:
I can imagine a future where we separately define the legacy and currently used operators, but that seemed like could be easily done in future PRs.

Reply:
Yeah, that seems OK.